src/tribler-core/tribler_core/restapi/tests/test_debug_endpoint.py

```diff
@@ -82,61 +83,104 @@ async def test_dump_memory(enable_api, session):
 
     # Check number of logs returned is correct
-    assert len(logs) == max_lines
+    assert len(logs) == num_logs
 
     # Check if test log message is present in the logs, at least once
     log_exists = any((True for log in logs if test_core_log_message in log))
     assert log_exists, "Test log not found in the debug log response"
 
 
+@pytest.mark.asyncio
+async def test_debug_pane_core_logs_in_root_dir(tribler_root_dir, enable_api, session):
+    """
+    Test whether the API returns the logs when logs are present in the root directory.
+    """
+
+    # Tribler logs are by default set to root state directory. Here we define the
+    # root state directory by updating 'TSTATEDIR' environment variable.
+    os.environ['TSTATEDIR'] = str(tribler_root_dir)
+    log_dir = get_root_state_directory()
+
+    process = 'core'
+    num_logs = 100
+
+    create_dummy_logs(log_dir, process='core', num_logs=num_logs)
+
+    json_response = await do_request(session, f'debug/log?process={process}&max_lines={num_logs}', expected_code=200)
+    logs = json_response['content'].strip().split("\n")
+
+    # Check number of logs returned is correct
+    assert len(logs) == num_logs
+
+
 @pytest.mark.asyncio
 async def test_debug_pane_default_num_logs(enable_api, session):
     """
     Test whether the API returns the last 100 logs when no max_lines parameter is not provided
     """
-    test_core_log_message = "This is the gui test log message"
-    expected_num_lines = 100
+    module = 'gui'
+    default_num_logs_returned = 100
+    num_logs_to_write = 200
 
     # Log directory
     log_dir = session.config.get_log_dir()
-    if not log_dir.exists():
-        os.makedirs(log_dir)
-    gui_info_log_file_path = log_dir / 'tribler-gui-info.log'
-
-    # write 200 (greater than expected_num_lines) test logs in file
-    with open(gui_info_log_file_path, "w") as core_info_log_file:
-        for log_index in range(200):   # write more logs
-            core_info_log_file.write("%s %d\n" % (test_core_log_message, log_index))
+    create_dummy_logs(log_dir, process=module, log_message=log_dir, num_logs=num_logs_to_write)
 
-    json_response = await do_request(session, 'debug/log?process=gui&max_lines=', expected_code=200)
+    json_response = await do_request(session, f'debug/log?process={module}&max_lines=', expected_code=200)
     logs = json_response['content'].strip().split("\n")
-    assert len(logs) == expected_num_lines
+    assert len(logs) == default_num_logs_returned
+
+
+@pytest.mark.asyncio
+async def test_debug_pane_no_logs(tribler_root_dir, enable_api, session):
+    """
+    Test whether the API returns the default response when no log files are found.
+    """
+    # To make sure that the test is not taking the OS dependent root state directory, we
+    # explicitly set the state directory here but we don't create any logs files there.
+    os.environ['TSTATEDIR'] = str(tribler_root_dir)
```

Comment:
Would this environment variable persist and leak to other tests?

Reply:
I don't think it does. I have set that in multiple tests which sets the value to `tribler_root_dir` which itself is different for different tests and it seems to work fine.

Reply:
I'm not entirely sure if this would also hold when running the tests with `pytest-xunit`, as we do in the PR testers. Can you double-check this and make sure it does not cause nasty, hidden issues?

Reply:
to be safe, now state directory environment variable is reset at the end

Reply:
@devos50 , thanks for catching this. I almost missed it.